File: billing_library/src/main/java/com/robj/billinglibrary/SkuDetails.java
package com.robj.billinglibrary;

/**
 * Created by Rob J on 10/02/18.
 */

public class SkuDetails {

    private final String sku;
    private final String type;
    private final String price;
    private final long priceAmountMicros;
    private final String priceCurrencyCode;
    private final String title;
    private final String description;
    private final String subscriptionPeriod;
    private final String freeTrialPeriod;
    private final String introductoryPrice;
    private final String introductoryPriceAmountMicros;
    private final String introductoryPricePeriod;
    private final String introductoryPriceCycles;
    private final com.android.billingclient.api.SkuDetails skuDetails;

    SkuDetails(com.android.billingclient.api.SkuDetails skuDetails) {
        this.skuDetails = skuDetails;
        sku = skuDetails.getSku();
        type = skuDetails.getType();
        price = skuDetails.getPrice();
        priceAmountMicros = skuDetails.getPriceAmountMicros();
        priceCurrencyCode = skuDetails.getPriceCurrencyCode();
        title = skuDetails.getTitle();
        description = skuDetails.getDescription();
        subscriptionPeriod = skuDetails.getSubscriptionPeriod();
        freeTrialPeriod = skuDetails.getFreeTrialPeriod();
        introductoryPrice = skuDetails.getIntroductoryPrice();
        introductoryPriceAmountMicros = skuDetails.getIntroductoryPriceAmountMicros();
        introductoryPricePeriod = skuDetails.getIntroductoryPricePeriod();
        introductoryPriceCycles = skuDetails.getIntroductoryPriceCycles();
    }

    public com.android.billingclient.api.SkuDetails getOriginalSkuDetails() {
        return skuDetails;
    }

    public String getSku() {
        return sku;
    }

    public String getType() {
        return type;
    }

    public String getPrice() {
        return price;
    }

    public long getPriceAmountMicros() {
        return priceAmountMicros;
    }

    public String getPriceCurrencyCode() {
        return priceCurrencyCode;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getSubscriptionPeriod() {
        return subscriptionPeriod;
    }

    public String getFreeTrialPeriod() {
        return freeTrialPeriod;
    }

    public String getIntroductoryPrice() {
        return introductoryPrice;
    }

    public String getIntroductoryPriceAmountMicros() {
        return introductoryPriceAmountMicros;
    }

    public String getIntroductoryPricePeriod() {
        return introductoryPricePeriod;
    }

    public String getIntroductoryPriceCycles() {
        return introductoryPriceCycles;
    }

    public com.android.billingclient.api.SkuDetails getSkuDetails() {
        return skuDetails;
    }
}

File: billing_library/src/main/java/com/robj/billinglibrary/models/SkuDetails.kt
package com.robj.billinglibrary.models

/**
 * Created by Rob J on 10/02/18.
 */

class SkuDetails internal constructor(val originalSkuDetails: com.android.billingclient.api.SkuDetails) {

    val sku: String
    val type: String
    val price: String
    val priceAmountMicros: Long
    val priceCurrencyCode: String
    val title: String
    val description: String
    val subscriptionPeriod: String
    val freeTrialPeriod: String
    val introductoryPrice: String
    val introductoryPriceAmountMicros: String
    val introductoryPricePeriod: String
    val introductoryPriceCycles: String

    init {
        sku = originalSkuDetails.sku
        type = originalSkuDetails.type
        price = originalSkuDetails.price
        priceAmountMicros = originalSkuDetails.priceAmountMicros
        priceCurrencyCode = originalSkuDetails.priceCurrencyCode
        title = originalSkuDetails.title
        description = originalSkuDetails.description
        subscriptionPeriod = originalSkuDetails.subscriptionPeriod
        freeTrialPeriod = originalSkuDetails.freeTrialPeriod
        introductoryPrice = originalSkuDetails.introductoryPrice
        introductoryPriceAmountMicros = originalSkuDetails.introductoryPriceAmountMicros
        introductoryPricePeriod = originalSkuDetails.introductoryPricePeriod
        introductoryPriceCycles = originalSkuDetails.introductoryPriceCycles
    }

    fun getSkuDetails(): com.android.billingclient.api.SkuDetails {
        return originalSkuDetails
    }
}
